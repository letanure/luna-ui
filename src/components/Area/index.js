import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

/**
 * Area component.
 * E.g.:
 * ```html
 * <Area>
 *  <div>
 *    ...
 *  </div>
 * </Area>
 * ```
 */
class AreaComponent extends React.PureComponent {
  render() {
    const { kind, className } = this.props;
    const Area = styled.div`
      padding: 20px;
      margin-bottom: 30px;
      box-shadow: 0 0 1px 1px rgba(145, 145, 145, .2);
      -webkit-box-shadow: 0 0 10px 2px rgba(145,145,145,.2);
      background-color: ${(props) => props.theme.areaColors[kind]};
    `;

    return <Area className={className}>{this.props.children}</Area>;
  }
}

AreaComponent.propTypes = {
  kind: PropTypes.string,
  className: PropTypes.string,
  children: PropTypes.node
};

AreaComponent.defaultProps = {
  kind: 'bg',
  className: '',
  children: null
};

export default AreaComponent;
