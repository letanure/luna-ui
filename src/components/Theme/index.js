import React from 'react';
import PropTypes from 'prop-types';
import { ThemeProvider } from 'styled-components';

import defaultTheme from '../../config/defaultTheme';
import DefaultStyles from '../../config/defaultStyles';
import DefaulTyphography from '../../config/defaultTyphography';

/**
 * Theme component.
 * E.g.:
 * ```html
 * <Theme theme={{...}}>
 *  ...
 * </Theme>
 * ```
 */
class ThemeComponent extends React.PureComponent {
  render() {
    const { theme, children } = this.props;

    return (
      <ThemeProvider theme={theme}>
        <div>
          <DefaultStyles />
          <DefaulTyphography />
          {children}
        </div>
      </ThemeProvider>
    );
  }
}

ThemeComponent.propTypes = {
  theme: PropTypes.object,
  children: PropTypes.node
};

ThemeComponent.defaultProps = {
  theme: defaultTheme,
  children: null
};

export default ThemeComponent;
