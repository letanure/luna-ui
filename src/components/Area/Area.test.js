import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import Theme from '../../config/defaultTheme';
import ThemeProvider from '../Theme/index';

import Area from './index';

describe('<Area />', () => {
  const component =
    <ThemeProvider theme={Theme}>
      <Area><strong>Lebron James has scored!</strong></Area>
    </ThemeProvider>;

  test('Render component', () => {
    const area = shallow(component);

    expect(area.render().find('strong')).toHaveLength(1);
    expect(area.render().text()).toEqual('Lebron James has scored!');
  });

  test('toMatchSnapshot', () => {
    const tree = renderer
      .create(component)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
