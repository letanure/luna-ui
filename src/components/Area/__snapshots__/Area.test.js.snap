// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Area /> toMatchSnapshot 1`] = `
<div>
  <div
    className="sc-htpNat fdTvEY"
  >
    <strong>
      Lebron James has scored!
    </strong>
  </div>
</div>
`;
